### DISCUSSION TITLE
Error during titanium build

### DISCUSSION THREAD

**Discussion by RomanLysiukOSF** [appcelerator/titanium_mobile#13310] [Q&A]
Hello everyone!

As you know, after March 1 Appcelerator Studio and the Appcelerator CLI aren't working.
So after reading [article from Ewan Harris](https://github.com/m1ga/from_zero_to_app/blob/master/appc_to_ti.md) I tried to build the app by `ti` command but it failed:

```
[DEBUG] Titanium SDK iOS directory: /Users/admin/Library/Application Support/Titanium/mobilesdk/osx/7.5.1.GA/iphone
[INFO]  Deploy type: development
[INFO]  Building for target: simulator
[INFO]  Building using iOS SDK: 12.1
[INFO]  Building for iOS Simulator: iPad Pro (12.9-inch) (3rd generation)
[DEBUG] UDID: B1AC9ADB-CE57-484E-BC59-4E5E10E7DC11
[DEBUG] Simulator type: ipad
[DEBUG] Simulator version: 12.1
[INFO]  Building for device family: ipad
[DEBUG] Setting Xcode target to Debug
[DEBUG] Setting Xcode build OS to iphonesimulator
[DEBUG] Xcode installation: /Applications/Xcode.app/Contents/Developer
[DEBUG] iOS WWDR certificate: installed
[INFO]  Team ID: n/a
[INFO]  Minimum iOS version: 8.0
[INFO]  Logging enabled on port 11472
[INFO]  Debugging disabled
[INFO]  Profiler disabled
[INFO]  Set to copy files instead of symlinking
[INFO]  Forcing clean build: /Users/admin/Repos/zmendlessaisle/build/iphone/build-manifest.json does not exist
[INFO]  Initializing the build directory
[DEBUG] Recreating /Users/admin/Repos/zmendlessaisle/build/iphone
[INFO]  Found Alloy app in /Users/admin/Repos/zmendlessaisle/app
[INFO]  Executing Alloy compile: /Users/admin/.nvm/versions/node/v8.9.4/bin/node  compile /Users/admin/Repos/zmendlessaisle/app --config platform=ios,version=0,simtype=none,devicefamily=ipad,deploytype=development,target=simulator
[DEBUG] module.js:540
[DEBUG]     throw err;
[DEBUG]     ^
[ERROR] : Cannot find module '/Users/admin/Repos/zmendlessaisle/undefined'
[DEBUG]     at Function.Module._resolveFilename (module.js:538:15)
[DEBUG]     at Function.Module._load (module.js:468:25)
[DEBUG]     at Function.Module.runMain (module.js:684:10)
[DEBUG]     at startup (bootstrap_node.js:187:16)
[DEBUG]     at bootstrap_node.js:608:3
[ERROR] Alloy compiler failed
```

This project was built using `appc` successfully but can't be built by `ti`. The error is quite strange because I don't even have any file with the name `bootstrap_node.js` (I did search over all filesystem).

System configuration:
```
Operating System
  Name                        = Mac OS X
  Version                     = 10.15.7
  Architecture                = 64bit
  # CPUs                      = 8
  Memory                      = 17179869184

Node.js
  Node.js Version             = 8.9.4
  npm Version                 = 5.6.0

Titanium CLI
  CLI Version                 = 5.2.0

Titanium SDK
  SDK Version                 = 7.5.1.GA
  SDK Path                    = /Users/admin/Library/Application Support/Titanium/mobilesdk/osx/7.5.1.GA
  Target Platform             = iphone
```
Has anyone faced with this or similar issue?

Thank you,
Roman

**Comment by m1ga** [comment]
Hi,
`node/v8.9.4` is the culprit. Please update to node 14 and reinstall the CLI. E.g. check my other tutorials about the installation at https://github.com/m1ga/from_zero_to_app/blob/master/installation.md#nodejs

**Comment by RomanLysiukOSF** [comment]
Hello @m1ga 
Thank you for your reply
Unfortunately our project is quite old and using SDK v7.5.1 that's why it needs Xcodebuild v10.1 and Node.JS v10 or previous. Actually I've tested with many versions of Node.JS but nothing changed. The only difference that on latest version of Node.JS we get an issue with "node-gyp" (because of MacOS Catalina), so build is failed even faster.

I also tried to create empty project on titanium and it works even on current configuration (Catalina, SDK 7.5.1, node v8.9.4, etc.). So the issue should be related to the project but I don't have any idea how to detect the problem.

And thanks for your tutorial!

Regards,
Roman

**Reply by m1ga** [reply]
think you won't be able to submit an update to the store with that SDK if its not an internal app. Which alloy version do you have installed? BTW: some people in slack said that `appc` still works for them so you could try that. 

but 7.5.1 is so old you should really update to be able to use a newer toolchain. Try 9.3.1 for example. Then you can use node 12.

**Reply by RomanLysiukOSF** [reply]
alloy package version is 1.17.3
the app is very old so I don't think I can migrate to new SDK without issues but I'll try at least

> BTW: some people in slack said that appc still works for them so you could try that.


on my side `appc` doesn't work:
```
admin@Admins-MacBook-Pro zmendlessaisle % appc run -p ios
Appcelerator Command-Line Interface, version 7.1.2
Copyright (c) 2014-2022, Appcelerator, Inc.  All Rights Reserved.

Appcelerator Login required to continue ...

? Appcelerator ID: roman.lysiuk@osf.digital
? Password: [hidden]
Invalid username or password.

2022-03-16T08:19:22.660Z | ERROR  | Invalid username or password.
```
login and password are valid of course

**Reply by jasonkneen** [reply]
Really need to bite the bullet and update to 10.1.X and then move forward. Drop me a DM on slack or email jason at bouncingfish.com and I'll see if I can help.

**Reply by caspahouzer** [reply]
For your appc problem: https://github.com/m1ga/from_zero_to_app/blob/master/appc_to_ti.md

**Reply by m1ga** [reply]
I would say the same. Update your app. If it is a store app you have to do it anyways.
Your Alloy version is new, your titanium version old, your SDK very old and your node very old. That combination doesn't fit very well. Even with the appc CLI there wouldn't be any support for that old SDK :smile: 

As Jason said: join us on tislack.org. There is a #job section and a #helpme
